api/endpointsServices/BaseService.js

```diff
@@ -0,0 +1,77 @@
+class BaseService {
+
+  constructor(schema) {
+    this.schema = schema;
+  }
+
+  insert(newSchema) {
+    return new Promise((resolve, reject) => {
+      newSchema.save(function (err) {
+        if (err) {
+          reject(err);
+        }
+        else {
+          resolve();
+        }
+      });
+    });
+
+  }
+
+  find(query) {
+    return new Promise((resolve, reject) => {
+      this.schema.find(query, function (err, results) {
+        if (err) {
+          reject(err);
+        }
+        else {
+          resolve(results);
+        }
+      });
+    });
+  }
+
+  update(query, update) {
+    return new Promise((resolve, reject) => {
+      this.schema.findOneAndUpdate(query, update, function (err, user) {
```

Comment:
why we have "user" in the base service ?

Reply:
fixed